### PR TITLE
Resolution for #108

### PR DESCRIPTION
**Pull request by spt1224**
Add getFullName() on the UserGroups model to fix Volunteers list when user is soft-deleted.
Remove group associations when soft deleting a user.

### CODE REVIEW

File: app/UserGroups.php [hunk 1]
```diff
@@ -93,4 +93,13 @@ public function scopeConfirmedInvitation($query)
     {
         return $query->where('status', 1)->orWhereNull('status');
     }
+
+    public function getFullName()
```
Comment (ngm) [inline app/UserGroups.php]: Is this called anywhere?

Comment (spt1224) [reply app/UserGroups.php]: Yes, but perhaps not anymore. These calls were the cause of the error that appeared when a user was soft deleted:

https://github.com/TheRestartProject/restarters.net/blob/master/resources/views/partials/volunteer-row.blade.php#L30
https://github.com/TheRestartProject/restarters.net/blob/master/resources/views/partials/volunteer-badge.blade.php#L27

Comment (ngm) [reply app/UserGroups.php]: Ah OK, bit confused as from diff it looks as if it's a completely new method.

**Comment by spt1224** [comment]
Well it is a completely new method, but it was being called. Maybe
$volunteer used to be a User instance rather than UserGroup?

**Pull request merged**